plugin-pinery/src/main/java/ca/on/oicr/gsi/shesmu/pinery/PinerySource.java

```diff
@@ -132,13 +132,21 @@ private Stream<PineryIUSForAnalysisValue> lanes(
                     lp.getLastModified() == null ? Instant.EPOCH : lp.getLastModified().toInstant();
                 addLane.accept(lp.getSequencerRunName(), lp.getLaneNumber());
 
+                Set<RunDtoContainer> containers = run.getContainers();
+                if (null != containers && containers.size() > 1) {
+                  multipleContainers.labels(run.getName()).inc();
+                  return null;
+                }
                 return new PineryIUSForAnalysisValue(
                     Optional.empty(),
                     run.getRunBasesMask() == null ? "" : run.getRunBasesMask(),
                     Set.of(),
                     Optional.empty(),
                     Optional.ofNullable(lp.getCreatedDate()).map(ZonedDateTime::toInstant),
-                    maybeGetRunField(run, RunDto::getContainerModel),
+                    null != containers && !containers.isEmpty()
+                        ? Optional.of(
+                            run.getContainers().stream().findFirst().get().getContainerModel())
```

Comment:
It shouldn't matter for our purposes, but you might want to use `Optional.ofNullable`. Container model should never be null from Pinery-MISO, but it's not strictly required in Pinery.